pokemon-app/src/components/EnemyPokemon.js

```diff
@@ -2,101 +2,46 @@ import React, { useState, useEffect } from "react";
-      ];
-
-    // Return it as an object of these four properties with the values as an array
-    // (this is so we could check if the enemy pokemon's move(s) are super
-    // effective, ineffective, or neutral against the friendly pokemon).
-    return {
-      strongAgainst: strongAgainst.split(", "),
-      weakAgainst: weakAgainst.split(", "),
-      resistantTo: resistantTo.split(", "),
-      vulnerableTo: vulnerableTo.split(", "),
-    };
-  });
-
-  // This function returns an array of integers. It gets the enemy pokemon's
-  // moves and creates a counter based on if it is effective against the
-  // friendly pokemon or not.
-  const moveEffectivenessCounter = enemyPokeMoveTypes.map((moveType) => {
-    let effectiveCounter = 0;
-
-    friendlyTypeMatchups.forEach((typeMatchups) => {
-      // If the enemy's move type is not effective against the friendly pokemon,
-      // then decrement.
-      if (typeMatchups.strongAgainst.includes(capitalize(moveType.name))) {
-        effectiveCounter -= 2;
-        // If the enemy's move type is effective against the friendly pokemon,
-        // then increment.
-      } else if (typeMatchups.weakAgainst.includes(capitalize(moveType.name))) {
-        effectiveCounter += 2;
-        // If the enemy's move type is a standard move type, then add 1.
-      } else {
-        effectiveCounter += 1;
-      }
-    });
-
-    return effectiveCounter;
-  });
-
-  // This returns the array of strings based on the effectiveness counter.
-  return moveEffectivenessCounter.map((effectiveCounter) => {
-    if (effectiveCounter >= 2) {
-      return "Super Effective";
-    } else if (effectiveCounter <= 0) {
-      return "Ineffective";
-    }
-
-    return "";
-  });
-};
+import {
+  moveEffectiveness,
+  overallEffectiveness,
+} from "../utils/effectiveness";
 
 const EnemyPokemon = ({
   enemyPokemon,
   hasEnemy,
   enemyPokeSoundUrl,
   enemyPokeMoveTypes,
   friendlyPokeType,
+  enemyPokeType,
 }) => {
-  const [enemyStatsOnTop, setEnemyStatsOnTop] = useState(true);
+  const [enemyStatsOnTop, setEnemyStatsOnTop] = useState(false);
   const [effectivenessArray, setEffectivenessArray] = useState([]);
+  const [overallEffectivenessString, setOverallEffectivenessString] =
+    useState("");
 
   // This checks if the moves property exist in pokemon object.
   const checkDoesMovesKeyExistInObject = () =>
     Object.prototype.hasOwnProperty.call(enemyPokemon, "moves");
 
   useEffect(() => {
-    if (enemyPokeMoveTypes && friendlyPokeType) {
+    if (
+      enemyPokeMoveTypes.length &&
+      friendlyPokeType.length &&
+      enemyPokeType.length
+    ) {
       setEffectivenessArray(
         moveEffectiveness(friendlyPokeType, enemyPokeMoveTypes)
       );
+
+      setOverallEffectivenessString(
+        overallEffectiveness(friendlyPokeType, enemyPokeType)
+      );
     }
-  }, [enemyPokeMoveTypes, friendlyPokeType]);
+  }, [enemyPokeMoveTypes, friendlyPokeType, enemyPokeType]);
 
```

Comment:
I really like this useEffect. Makes me see more ways to use them than I currently do. Great work.